book/course/spack/existing.md

```diff
@@ -0,0 +1,244 @@
+                ^bison@3.8.2%gcc@8.5.0 arch=linux-rhel8-haswell
+                ^gettext@0.21%gcc@8.5.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-rhel8-haswell
+                    ^tar@1.34%gcc@8.5.0 zip=pigz arch=linux-rhel8-haswell
+                        ^pigz@2.7%gcc@8.5.0 arch=linux-rhel8-haswell
+                        ^zstd@1.5.2%gcc@8.5.0+programs compression=none libs=shared,static arch=linux-rhel8-haswell
+            ^libedit@3.1-20210216%gcc@8.5.0 arch=linux-rhel8-haswell
+        ^pmix@4.1.2%gcc@8.5.0~docs+pmi_backwards_compatibility~restful arch=linux-rhel8-haswell
+            ^libevent@2.1.12%gcc@8.5.0+openssl arch=linux-rhel8-haswell
+```
+
+</details>
+
+You don't need to read the full spec above, but on my system, this adds up to
+36 packages, which I'm sure would take a fair old amount of time to build.  So
+let's see if we can cut that down by using software already installed on my
+system.
+
+It can find certain software in your environment without you having to
+configure anything by hand.  So if I load an openmpi module, and run:
+
+```bash
+$ spack external find openmpi
+==> The following specs have been detected on this system and added to /home/me/.spack/packages.yaml
+-- no arch / gcc@8.5.0 ------------------------------------------
+openmpi@4.1.1
+```
+
+Spack is now aware of that software, so you can build against it without having
+to rebuild openmpi, as long as you're happy with that version.  As you can see
+from the output, this is written into your configuration, so will be remembered
+for future runs.
+
+I also want it to discover cmake:
+
+```bash
+$ spack external find cmake
+==> The following specs have been detected on this system and added to /home/me/.spack/packages.yaml
+cmake@3.20.2
+```
+
+If Spack wasn't able to auto detect the software, you can add it to the
+packages.yaml by hand.  For example, if I wanted to add a system installed
+libjpeg-turbo to spack, I can add a section for that into my packages.yaml:
+
+```yaml
+  libjpeg-turbo:
+    externals:
+    - spec: libjpeg-turbo@1.5.3
+      prefix: /usr
+```
+
+My full packages.yaml now looks like this, if looked at with `spack config edit packages`:
+
+```yaml
+packages:
+  openmpi:
+    externals:
+    - spec: openmpi@4.1.1%gcc@8.5.0+atomics~cuda+cxx~cxx_exceptions+java~memchecker+pmi~static~wrapper-rpath
+        fabrics=ofi,psm2,ucx schedulers=slurm
+      prefix: /usr/lib64/openmpi
+  cmake:
+    externals:
+    - spec: cmake@3.20.2
+      prefix: /usr
+  all:
+    compiler:
+    - gcc@8.3.0   # Your preferred compiler here
+  libjpeg-turbo:
+    externals:
+    - spec: libjpeg-turbo@1.5.3
+      prefix: /usr
+```
+
+There's a whole lot of interesting stuff in there that I've not had to think
+about, as Spack has added all the details about the openmpi variant it
+discovered.  We'll cover some of this in the advanced building section.
+Annoyingly, it got one detail wrong because of how Red Hat split the
```

Comment:
This might warrant an explicit admonition-style callout as it's more a gotcha-hint than general tutorial material?